### PR TITLE
Fix Volt Switch / Roar with ace pokemon

### LINKED ISSUE #2591: AI_FLAG_ACE_POKEMON is currently broken
```
    [TRAINER_WATTSON_1] =
    {
        .trainerClass = TRAINER_CLASS_LEADER,
        .encounterMusic_gender = TRAINER_ENCOUNTER_MUSIC_MALE,
        .trainerPic = TRAINER_PIC_LEADER_WATTSON,
        .trainerName = _("Wattson"),
        .items = {ITEM_SUPER_POTION, ITEM_SUPER_POTION, ITEM_NONE, ITEM_NONE},
        .doubleBattle = FALSE,
        .aiFlags = AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_TRY_TO_FAINT | AI_FLAG_CHECK_VIABILITY | AI_FLAG_ACE_POKEMON,
        .party = ITEM_CUSTOM_MOVES(sParty_Wattson1),
    },
```

```
static const struct TrainerMonItemCustomMoves sParty_Wattson1[] = {
    {
    .iv = 200,
    .lvl = 22,
    .species = SPECIES_EMOLGA,
    .heldItem = ITEM_NONE,
    .moves = {MOVE_VOLT_SWITCH, MOVE_NONE, MOVE_NONE, MOVE_NONE},
    },
    {
    .iv = 250,
    .lvl = 24,
    .species = SPECIES_ZEBSTRIKA,
    .heldItem = ITEM_SITRUS_BERRY,
    .moves = {MOVE_QUICK_ATTACK, MOVE_SHOCK_WAVE, MOVE_VOLT_SWITCH, MOVE_FLAME_CHARGE},
    }
};
```

To reproduce:
-Make enemy team have 2 Pokémon alive, 1 + ACE
-Make the non ace Pokémon use volt switch (I assume it also happens with other switch moves/abilities/items)

### PR DESCRIPTION
**Pull request by DizzyEggg**
Fixes #2591

### CODE REVIEW

File: src/battle_ai_switch_items.c [hunk 1]
```diff
@@ -27,6 +27,15 @@ static bool32 AiExpectsToFaintPlayer(void);
 static bool32 AI_ShouldHeal(u32 healAmount);
 static bool32 AI_OpponentCanFaintAiWithMod(u32 healAmount);
 
+static bool32 isAceMon(u32 battlerId, u32 monPartyId)
```
Comment (ghoulslash) [inline src/battle_ai_switch_items.c:30]: `IsAceMon`

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: src/battle_ai_switch_items.c (hunk 1 above)
Comment (DizzyEggg) [reply src/battle_ai_switch_items.c]: Done

File: src/battle_ai_switch_items.c [hunk 2]
```diff
@@ -27,6 +27,15 @@ static bool32 AiExpectsToFaintPlayer(void);
 static bool32 AI_ShouldHeal(u32 healAmount);
 static bool32 AI_OpponentCanFaintAiWithMod(u32 healAmount);
 
+static bool32 IsAceMon(u32 battlerId, u32 monPartyId)
+{
+    if (AI_THINKING_STRUCT->aiFlags & AI_FLAG_ACE_POKEMON
+        && !(gBattleStruct->forcedSwitch)
```
Comment (mrgriffin) [inline src/battle_ai_switch_items.c:33]: Should this say `!(gBattleStruct->forcedSwitch & gBitTable[battlerId])`? (Maybe it shouldn't, I'm not too sure why there's a `forcedSwitch` bit for each battler.)

Comment (DizzyEggg) [reply src/battle_ai_switch_items.c]: Yeah, you're right, I fixed it

**Review by Jaizu — APPROVED** [review]

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**